torchci/components/DarkModeEChart.tsx

```diff
@@ -0,0 +1,193 @@
+
+export default function DarkModeEChart(props: any) {
+  const { darkMode } = useDarkMode();
+  const chartRef = useRef(null);
+
+  // Function to update all text colors in the chart SVG
+  const updateChartColors = () => {
+    if (!chartRef.current) return;
+
+    // Access the DOM elements directly
+    const chartElement = chartRef.current as any;
+    if (!chartElement || !chartElement.getEchartsInstance) return;
+
+    try {
+      const echartsInstance = chartElement.getEchartsInstance();
+      const dom = echartsInstance.getDom();
+      if (!dom) return;
+
+      const textColor = darkMode ? "#E0E0E0" : "#212529";
+
+      // Update all text elements
+      dom.querySelectorAll("text").forEach((text: SVGTextElement) => {
+        text.style.fill = textColor;
+        text.setAttribute("fill", textColor);
+      });
+
+      // Special handling for title text
+      dom.querySelectorAll(".ec-title, .ec-title-sub").forEach((el: any) => {
+        if (el && el.style) {
+          el.style.fill = textColor;
+          el.style.color = textColor;
+        }
+
+        // Also look for text elements inside titles
+        el.querySelectorAll("text, tspan").forEach((text: SVGTextElement) => {
+          text.style.fill = textColor;
+          text.setAttribute("fill", textColor);
+        });
+      });
+
+      // Update SVG title and subtitle elements
+      dom
+        .querySelectorAll(
+          '[data-zr-dom-id="zr_0"] title, [data-zr-dom-id="zr_0"] title-sub'
+        )
+        .forEach((el: any) => {
+          if (el && el.style) {
+            el.style.fill = textColor;
+            el.setAttribute("fill", textColor);
+          }
+        });
+    } catch (e) {
+      console.error("Failed to update chart colors:", e);
+    }
+  };
+
+  // Update colors whenever the chart is rendered or dark mode changes
+  useEffect(() => {
+    // Apply multiple times to ensure it catches all text elements
+    const timers = [
+      setTimeout(() => updateChartColors(), 100),
+      setTimeout(() => updateChartColors(), 500),
+      setTimeout(() => updateChartColors(), 1000),
+    ];
+
+    // Set up event listener for chart rendering
+    if (chartRef.current) {
+      const chartElement = chartRef.current as any;
+      if (chartElement.getEchartsInstance) {
+        const echartsInstance = chartElement.getEchartsInstance();
+        echartsInstance.on("rendered", updateChartColors);
+        echartsInstance.on("finished", updateChartColors);
+
+        // Direct hack for title and subtitle
+        echartsInstance.on("finished", () => {
+          try {
+            // Force update title text
+            const titleElements = chartElement.ele?.querySelectorAll(
+              ".ec-title, .ec-title-sub"
+            );
+            const textColor = darkMode ? "#E0E0E0" : "#212529";
+
+            if (titleElements && titleElements.length > 0) {
+              titleElements.forEach((el: any) => {
```

Comment:
I'm not a typescript programmer, but why do we care to skip the loop for empty arrays, it will simply never gets caught
```suggestion
            
              titleElements?.forEach((el: any) => {
```